explainability-core/src/main/java/org/kie/trustyai/explainability/model/Dataframe.java

```diff
@@ -0,0 +1,688 @@
+package org.kie.trustyai.explainability.model;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.kie.trustyai.explainability.model.domain.EmptyFeatureDomain;
+import org.kie.trustyai.explainability.model.domain.FeatureDomain;
+import org.kie.trustyai.explainability.model.domain.NumericalFeatureDomain;
+import org.kie.trustyai.explainability.model.domain.ObjectFeatureDomain;
+
+public class Dataframe {
+
+    private List<List<Value>> data = new ArrayList<>();
+    private Metadata metadata = new Metadata();
+
+    private Dataframe() {
+
+    }
```

Comment:
We can probably make them `final` 

```suggestion
    private final List<List<Value>> data;
    private final Metadata metadata;

    private Dataframe() {
        this(new ArrayList<>(), new Metadata());
    }
    
    private Dataframe(List<List<Value>> data, Metadata metadata) {
        this.data = data;
        this.metadata = metadata;
    }
```